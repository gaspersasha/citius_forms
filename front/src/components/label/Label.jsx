import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { INPUT_STATUS } from '~constants';
import s from './label.module.sass';

const Label = ({ title, customClasses, name, children, status, withSub }) => (
  <div
    className={cn(s.label, customClasses, {
      'error-input-label': status === INPUT_STATUS.INVALID,
    })}
  >
    <label
      htmlFor={name || ''}
      data-id={status === INPUT_STATUS.INVALID && 'with-error'}
    >
      <span
        className={cn(s.title, withSub && s.halfMargin)}
        data-id="label-title"
      >
        {title}
      </span>
      {children}
    </label>
  </div>
);

Label.propTypes = {
  customClasses: PropTypes.string,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  name: PropTypes.string,
  children: PropTypes.any,
  status: PropTypes.string,
  withSub: PropTypes.bool,
};

Label.defaultProps = {
  title: '',
  customClasses: '',
  name: '',
  children: null,
  status: INPUT_STATUS.DEFAULT,
  withSub: false,
};

export default Label;
