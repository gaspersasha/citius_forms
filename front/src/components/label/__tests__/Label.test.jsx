import React from 'react';
import Label from '../Label';

describe('<Label />', () => {
  let wrapper;

  const getWrapper = (props) => shallow(<Label {...props} />);

  it('renders without errors', () => {
    wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});
