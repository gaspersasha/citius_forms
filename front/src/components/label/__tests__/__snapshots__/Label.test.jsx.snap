// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> renders without errors 1`] = `
<div
  className="label"
>
  <label
    data-id={false}
    htmlFor=""
  >
    <span
      className="title"
      data-id="label-title"
    />
  </label>
</div>
`;
